File: 2020/day1/javascript/part2.js
// See: https://nodejs.dev/learn/introduction-to-nodejs

fs = require('fs');

const data = fs.readFileSync("../input.txt", 'utf8');

lines = data.split("\n");

let numbers = [];
for (let i in lines) {
    const line = lines[i];
    numbers.push(parseInt(line));
}

for (let i=0; i<numbers.length; i++) {
    for (let j=i+1; j<numbers.length; j++) {
        for (let k=j+1; k<numbers.length; k++) {
            if (numbers[i] + numbers[j] + numbers[k] == 2020) {
                console.log(numbers[i]*numbers[j]*numbers[k]);
            }
        }
    }
}

File: 2020/day1/c#/part2.cs
using System;

public class Part1
{
    public static void Main(string[] args)
    {
        string[] lines = System.IO.File.ReadAllLines(@"../input.txt");

        int[] numbers = new int[lines.Length];
        int idx=0;
        foreach (string line in lines) {
            numbers[idx++] = int.Parse(line);
        }

        for (int i = 0; i < numbers.Length; i++) {
            int n = numbers[i];
            for (int j = i+1; j < numbers.Length; j++) {
                int m = numbers[j];
                for (int k = j+1; k < numbers.Length; k++) {
                    int o = numbers[k];
                    if (n + m + o== 2020) {
                        Console.WriteLine(n*m*o);
                    }
                }
            }
        }
    }
}
